Fix work types not showing as options with Project Extent - Generic

mddilley opened the pull request:
## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/19133

## Testing
**URL to test:** 
<!---
 [branch-name]--atd-moped-main.netlify.app/moped/ if test instance or deploy-preview-[pr-number]--atd-moped-main.netlify.app/moped/ 
--->
Deploy preview

**Steps to test:**
1. Create a new component and choose the **Project Extent - Generic** component type
2. Try to select a work type and you should see three options - New, Maintenance / Repair, and Modification
3. Try the same in staging and you should see that no options show up in the work type dropdown
4. Check that subcomponents still work too since I updated that line to avoid this same type of bug

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Product manager checked [DB view dependencies](https://atd-dts.gitbook.io/moped-documentation/product-management/updating-the-arcgis-online-components-database-view)
- [ ] Product manager added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

mddilley commented on moped-editor/src/views/projects/projectView/ProjectComponents/utils/form.js:
```diff
@@ -99,7 +99,7 @@ export const useComponentOptionsWithoutSignals = (options) =>
 export const useSubcomponentOptions = (componentId, optionsData) =>
   useMemo(() => {
-    if (!componentId || !optionsData) return [];
+    if ((!componentId && componentId !== 0) || !optionsData) return [];
```
Hopefully this help us remember that 0 is a valid id 🤔

johnclary reviewed (approved):
it's passing the 👀 test for me ✅ 🚢 

i wonder how disastrous it would be to (in a separate PR) change this component ID to be non-zero 🤔 it's just a little gross

chiaberry reviewed (approved):
tested and now i can pick work types

🚢

mddilley commented:
Added the version bump in here so it makes its way into `main`. Merging!

@johnclary I agree that would be some good tech debt to address someday and remove the yuckiness. We have a few other lookup tables where we inserted `0` ids through migrations long ago too. Just added to my list of issues to create. 🙏
